catalog/spatial/registry/registry-federation-admin-impl/src/test/java/org/codice/ddf/registry/federationadmin/impl/FederationAdminTest.java

```diff
@@ -800,6 +807,109 @@ public void testRegenerateSources() throws Exception {
+        assertSummary(mcardMap, timestamp);
+    }
+
+    @Test
+    public void testRegistryMetacard() throws Exception {
+
+        when(federationAdminService.getRegistryObjectByRegistryId(any())).thenReturn(
+                getRegistryObjectFromResource("/csw-full-registry-package.xml"));
+        ArrayList<String> tags = new ArrayList<>();
+        tags.add(RegistryConstants.REGISTRY_TAG);
+        mcard.setAttribute(Metacard.TAGS, tags);
+        mcard.setAttribute(RegistryObjectMetacardType.REGISTRY_ID,
+                "urn:uuid:2014ca7f59ac46f495e32b4a67a51276");
+
+        when(federationAdminService.getRegistryMetacardsByRegistryIds(any())).thenReturn(Collections.singletonList(
+                mcard));
+        List<Map<String, Object>> result =
+                (List<Map<String, Object>>) federationAdmin.registryMetacard(
+                        "urn:uuid:2014ca7f59ac46f495e32b4a67a51276")
+                        .get("nodes");
+        assertThat(result.size(), is(1));
+        Map<String, Object> mcardMap = result.get(0);
+        assertThat(mcardMap.get("id"), is("urn:uuid:2014ca7f59ac46f495e32b4a67a51276"));
+    }
+
+    @Test
+    public void testRegistryMetacardBadId() throws Exception {
+        when(federationAdminService.getRegistryObjectByRegistryId(any())).thenThrow(new FederationAdminException(
+                "Not found"));
+
+        when(federationAdminService.getRegistryMetacardsByRegistryIds(any())).thenReturn(Collections.emptyList());
+        List<Map<String, Object>> result =
+                (List<Map<String, Object>>) federationAdmin.registryMetacard(
+                        "urn:uuid:2014ca7f59ac46f495e32b4a67a51276")
+                        .get("nodes");
+        assertThat(result.size(), is(0));
+    }
+
+    private Date setupSummary() throws Exception {
+        Date now = new Date();
+        ArrayList<String> tags = new ArrayList<>();
+        tags.add(RegistryConstants.REGISTRY_TAG);
+        mcard.setAttribute(Metacard.TAGS, tags);
+        mcard.setTitle("TestTitle");
+        mcard.setCreatedDate(now);
+        mcard.setModifiedDate(now);
+        mcard.setAttribute(RegistryObjectMetacardType.REGISTRY_LOCAL_NODE, true);
+        mcard.setAttribute(RegistryObjectMetacardType.REGISTRY_IDENTITY_NODE, true);
+        when(federationAdminService.getRegistryMetacards()).thenReturn(Collections.singletonList(
+                mcard));
+        return now;
+    }
+
+    private void assertSummary(Map<String, Object> mcardMap, Date timestamp) {
+        assertThat(mcardMap.get("metacardId"), is("someUUID"));
+        assertThat(mcardMap.get("registryId"), is("myId"));
+        assertThat(mcardMap.get("name"), is("TestTitle"));
+        assertThat(mcardMap.get("created"), is(timestamp));
+        assertThat(mcardMap.get("modified"), is(timestamp));
+        assertThat(mcardMap.get("identityNode"), is(true));
+        assertThat(mcardMap.get("localNode"), is(true));
```

Comment:
These String values can be constants because they match the Strings used in the test setups.